src/mysky/index.ts

```diff
@@ -376,14 +401,92 @@ export class MySky {
     const dataKey = deriveDiscoverableTweak(path);
     opts.hashedDataKeyHex = true; // Do not hash the tweak anymore.
 
-    const entry = await getDeletionRegistryEntry(this.connector.client, publicKey, dataKey, opts);
+    const getEntryOpts = extractOptions(opts, defaultGetEntryOptions);
+    const entry = await getNextRegistryEntry(
+      this.connector.client,
+      publicKey,
+      dataKey,
+      new Uint8Array(RAW_SKYLINK_SIZE),
+      getEntryOpts
+    );
 
     const signature = await this.signRegistryEntry(entry, path);
 
     const setEntryOpts = extractOptions(opts, defaultSetEntryOptions);
     await this.connector.client.registry.postSignedEntry(publicKey, entry, signature, setEntryOpts);
   }
 
+  /**
+   * Gets the raw registry entry data for the given path, if the user has given READ permissions.
+   *
+   * @param path - The data path.
+   * @param [customOptions] - Additional settings that can optionally be set.
+   * @returns - The entry data.
+   */
+  async getEntryData(path: string, customOptions: CustomGetEntryOptions): Promise<EntryData> {
```

Comment:
I'm not sure why we wrap the data in `EntryData`. That seems useless, right? Especially seeing as `setEntryData` takes a `Uint8Array` so they don't mirror each other.

Reply:
It's the design decision we've made in the SDK so that we can add more fields later without a breaking change. Right now the only field is `data`:

```typescript
const { data } = getEntryData(...);
```

but later there may be more:

```typescript
const { data, foo } = getEntryData(...);
```

Reply:
What other field would `getEntryData` return in the future.

Reply:
We don't know yet how MySky and SkyDB will evolve over time. It's better to leave it open for more fields.